django_mongodb/client.py

```diff
@@ -4,11 +4,44 @@
 
 
 class DatabaseClient(BaseDatabaseClient):
-    executable_name = "mongo"
+    executable_name = "mongosh"
 
     @classmethod
     def settings_to_cmd_args_env(cls, settings_dict, parameters):
-        raise NotImplementedError
+        options = settings_dict.get("OPTIONS", {})
+        uri = options.get("uri")
+        args = [cls.executable_name]
+        if not uri:
+            host = settings_dict["HOST"]
+            port = settings_dict.get("PORT")
+            if port and host:
+                host += f":{port}"
+
+            dbname = settings_dict.get("NAME", "")
+            user = settings_dict.get("USER")
+            passwd = settings_dict.get("PASSWORD")
+            auth_database = options.get("authenticationDatabase")
+            auth_mechanism = options.get("authenticationMechanism")
+            retry_writes = options.get("retryWrites")
+            protocol = options.get("protocol", "mongodb")
+
+            if user:
+                args += ["--username", user]
+            if passwd:
+                args += ["--password", passwd]
+            if auth_database:
+                args += ["--authenticationDatabase", auth_database]
+            if auth_mechanism:
+                args += ["--authenticationMechanism", auth_mechanism]
+            if retry_writes is not None:
+                args += ["--retryWrites", str(bool(retry_writes)).lower()]
+            if host:
+                uri = f"{protocol}://{host}/{dbname}"
+
+        if uri:
+            args.append(uri)
+        args += options.get("filenames", []) + parameters
```

Comment:
I don't think `filenames` should be specified this way. It doesn't make sense to put that information in your Django config (and it would crash `MongoClient` since it's not a parameter). Note that if you run `manage.py dbshell abc def` the additional arguments are passed to `mongosh`, so I think it's already possible to provide filenames that way.

Reply:
Yes, could be. I will check this